Remotion/Data/DomainObjects.UnitTests/TestDomainDatabase.cs

```diff
@@ -0,0 +1,103 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Data.SqlClient;
+using Remotion.Data.DomainObjects.Mapping;
+using Remotion.Data.DomainObjects.UnitTests.Database;
+using Remotion.Data.DomainObjects.UnitTests.Factories;
+using Remotion.Development.UnitTesting;
+using Remotion.Development.UnitTesting.Data.SqlClient;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.UnitTests
+{
+  /// <summary>
+  /// Provides methods for setting up the test domain database for the integration tests.
+  /// The initialization takes some time so it is only executed if the tests demand it (see <see cref="DatabaseTest"/>).
+  /// </summary>
+  public static class TestDomainDatabase
+  {
+    private static readonly object s_lock = new();
+
+    private static bool s_initialized;
+    private static StandardMappingDatabaseAgent s_standardMappingDatabaseAgent;
+
+    public static void InitializeDatabase ()
+    {
+      if (s_initialized)
+        return;
+
+      lock (s_lock)
```

Comment:
why did you add a lockstatement here but not in the other helper types that where moved from setupfixture to other types.

Reply:
Because the database initialization takes a long time and it might be called from multiple threads if tests are parallelized. The `SetupFixture` had the implicit guarantee that it is called once at the start of the tests, but this guarantee is no longer there. Multiple inserts of the test data might cause weird test failures.